compute_tools/src/compute.rs

```diff
@@ -126,13 +156,14 @@ impl ComputeNode {
     // unarchive it to `pgdata` directory overriding all its previous content.
     #[instrument(skip(self, compute_state))]
     fn get_basebackup(&self, compute_state: &ComputeState, lsn: &str) -> Result<()> {
+        let spec = compute_state.pspec.as_ref().expect("spec must be set");
```

Comment:
NIT: Not sure how important it is in this code, but we can return `Err` instead of panicking, same for below.

Reply:
I consider it a pretty fundamental programmer error to call these functions without setting the spec first, so panic seems appropriate. Either way, the program will exit.